test/types/create.test.ts

```diff
@@ -9,7 +9,7 @@ interface ITest extends Document {
 
 const Test = model<ITest>('Test', schema);
 
-Test.create({ _id: '0'.repeat(24), name: 'test' }).then((doc: ITest) => console.log(doc.name));
+Test.create({ _id: new Types.ObjectId('0'.repeat(24)), name: 'test' }).then((doc: ITest) => console.log(doc.name));
```

Comment:
This change is a bit concerning. We should at least support passing ObjectIds as strings in `create()`, you shouldn't need the `new Types.ObjectId()` here.

Reply:
It supports passing object as string, but then it needs to use  type assertion to workaround ts error like so:
```
Test.create({ _id: '0'.repeat(24) as unknown as Types.ObjectId, name: 'test' }).then((doc: ITest) => console.log(doc.name));
```
Or like the following but doc type will be string instead of object id.
```
Test.create({ _id: '0'.repeat(24) as unknown as Types.ObjectId, name: 'test' }).then((doc) => console.log(doc.name));

```
else, ITest interface can be as following:

```
interface ITest extends Document {
  _id?: Types.ObjectId | string;
  name?: string;
}
```

Reply:
I think this is a serious regression, breaking user space, which must be avoided. The workarounds are not feasable, because mongoose internally casts the _id string to a ObjectId. So we have to ensure if _id is defined as  ObjectId that _id is defined as `_id?: Types.ObjectId | string`

Reply:
@Uzlopak  @vkarpov15 
I would like just to clarify some things here:
1- This specific example use enforced doc type which means It's not auto typed doc, Therefore I think the user should be the responsible to provide the correct type here. 
2- mongoose allows defining _id path as a number  as mentioned here https://mongoosejs.com/docs/guide.html#_id.
Therefore, I think it is wrong to make _id always as 'string' | 'object'  which will not be correct in this case.
3- we can make every path that is defined as 'Objected' in schema definition as string | ObjectId in auto typed schema, but then it will be ObjectId | string for all paths not just _id, so is that correct then ?